neo/public/javascripts/common/services.js

```diff
@@ -74,6 +74,18 @@ angular.module('cloudberry.common', [])
       ];
     }
 
+    function getPopulationTarget(parameters){
+      var targetLowerCase = parameters.geoLevel;
+      var targetUpperCase = targetLowerCase.charAt(0).toUpperCase() + targetLowerCase.substr(1);
+      return {
+        joinKey: [targetLowerCase],
+        dataset: "twitter.ds" + targetUpperCase + "Population",
```

Comment:
Not that complex. You may define a map or a meta object to have mappings like state->statePopulation, county->countyPopulation etc. This mapping should also be resided in the global config object (if we have one).